### ISSUE TITLE
`imagetools create` should return the digest of the new image

### ISSUE THREAD

**Issue by Pesa** [docker/buildx#2407]
### Description

I'm [using](https://github.com/named-data/actions/blob/2907fde08ddbfed9ca5284b517f888a029baeeeb/.github/workflows/docker-image.yml#L192) `docker buildx imagetools create` to create a multi-platform image from previously built single-platform images.

I'd like this command to somehow output the digest of the newly created image, so that subsequent steps and downstream jobs can "atomically" operate (run some checks, inspect, sign, etc.) on that same version of the image without relying on tags, which can be moved.

`docker/build-push-action` already has a `digest` output, so I guess I'm asking for a similar feature in `imagetools create`.

**Comment by tonistiigi** [comment]
@crazy-max Should we add `--metadata-file` to `imagetools create` or do you have better ideas for UX. We can also do it for `--dry-run` that atm. prints the manifest bytes. Digest is just checksum of the same bytes.

**Comment by crazy-max** [comment]
> @crazy-max Should we add `--metadata-file` to `imagetools create` or do you have better ideas for UX. We can also do it for `--dry-run` that atm. prints the manifest bytes. Digest is just checksum of the same bytes.

I think it would be enough to just print the digest imo.

**Comment by thompson-shaun** [comment]
Let's move ahead with the suggestion from @tonistiigi

**Comment by zanieb** [comment]
Just want to give a gentle poke here to say we need this feature over at Astral and am curious if anyone is working on it and / or if it's open for contributions?

**Comment by tonistiigi** [comment]
I think it would be weird to just return digest when `--dry-run` returns the whole manifest bytes. Other suggestions? Or is everyone else ok with that inconsistency?

**Comment by zanieb** [comment]
Is the "workaround" then to just hash the bytes yourself?

I think the alternative would be an opt-in JSON output format with the bytes and digest in a structure form?

**Comment by tonistiigi** [comment]
> Is the "workaround" then to just hash the bytes yourself?

Atm. I think `--dry-run` shows the manifest bytes. Without `--dry-run` there is no stdout output.

**Comment by Pesa** [comment]
> I think it would be weird to just return digest when `--dry-run` returns the whole manifest bytes. Other suggestions? Or is everyone else ok with that inconsistency?

`--dry-run` is specifically [documented](https://docs.docker.com/reference/cli/docker/buildx/imagetools/create/#options) to "show final image instead of pushing", so the potential inconsistency doesn't sound too worrisome to me. In other words, in the dry-run case, the user explicitly requested to print the manifest.

That being said, adding a new `--print-digest` flag would avoid the inconsistency altogether and is probably more future proof.

I also like @zanieb's suggestion of an opt-in JSON output.

**Comment by zanieb** [comment]
I agree the inconsistency with dry-run seems fine. I think outputting the digest without opt-in seems pretty reasonable, don't other Docker commands return an identifier for the object they create?

**Comment by tonistiigi** [comment]
Ok, let's output the digest on a successful push then.

**Comment by xlionjuan** [comment]
well, this issue is being forgotten.

**Comment by crazy-max** [comment]
Fixed in https://github.com/docker/buildx/pull/3638
